app/models/user.rb

```diff
@@ -81,20 +81,27 @@ class User < ApplicationRecord
   validates :locale, inclusion: {
     in: Rails.application.config.i18n.available_locales.map(&:to_s).push(nil)
   }
-  validate :password_complexity
+  validate :password_complexity, unless: :oauth_provided?
 
   def active_for_authentication?
     super && !banned
   end
 
-  def self.find_for_google_oauth2(access_token)
-    user = find_or_initialize_by(email: access_token.info.email)
-    user.name ||= access_token.info.name
+  def self.find_for_google_oauth2(auth)
```

Comment:
Is `auth` an object or a value(token?)?

Reply:
It's a token

Reply:
This is a good question Ben asked!

It seems like it's an object, but line 94 treats these entire object as a token. That seems off to me. I would double-check what the value of auth actually is.